### PR TITLE
Make Sure `KeyError` is Thrown

### PR DESCRIPTION
**Pull request by ilan-gold**
Just calling `zarr.open` yields a special `zarr` error so I think this is more robust.

### CODE REVIEW

**Review by mccalluc — CHANGES_REQUESTED** [review]
Thanks. Some stylistic suggestions... but skip the first two, and just look at the last: I'd suggest getting rid of the try-except.

This is related to https://github.com/hubmapconsortium/portal-ui/pull/2656 ?

File: src/portal_visualization/builders/anndata_builders.py [hunk 1]
```diff
@@ -55,7 +55,8 @@ def get_conf_cells(self):
         # Check for an alias for gene names to add to the view config.
         gene_alias = 'var/hugo_symbol'
         try:
-            zarr.open(f'{adata_url}/{gene_alias}', mode='r')
+            z = zarr.open(f'{adata_url}', mode='r')
+            z['var']['hugo_symbol']
         except KeyError:
             gene_alias = None
```
[the remark this answers is not in the document]
Comment (ilan-gold) [reply src/portal_visualization/builders/anndata_builders.py]: Ok!

**Comment by ilan-gold** [comment]
@mccalluc Re-requesting review but I noticed something:
Screen Shot 2022-05-24 at 8 05 42 PM
The marker genes are hardcoded ENSEMBL IDs.  We should ask Nils what he wants to do about this.

**Comment by ngehlenborg** [comment]
How are the marker genes hardcoded as Ensembl IDs? Do you meant they are provided only as Ensembl IDs?

If so, it should be possible to use the mapping in the annData store to look up the corresponding gene symbols. If there is no corresponding gene symbol, we should fall back on the Ensembl ID (or use whatever other mechanism we will have in place for Ensembl IDs that are not mapped to gene symbols.

**Comment by ilan-gold** [comment]
@ngehlenborg When I said hardcoded, I meant that we really can't translate them.  In theory we could map this, but we would need to create a mechanism in Vitessce that does "take factors i.e annotations of cells and if they are gene ids, try to look up the translation in the AnnData field we passed in" - I don't think this is something we want to implement but maybe @keller-mark has a better idea.

**Comment by ngehlenborg** [comment]
Ok. Let's leave the marker gene annotations as is but switch everything else over to gene symbols. Re-processing of RNA-seq data might never happen given the big backlog of unprocessed data.

**Review by mccalluc — COMMENTED** [review]
@ilan-gold -- Can you give me the UUID of a dataset I can try with this, and then flag me for review again? I'll use the CLI here to preview it -- I just don't want to be going back and forth between here and the portal.

**Comment by ilan-gold** [comment]
@mccalluc The two datasets you commented on in the issue should work great - one is missing the HUGO ID's (the first one) and the second has them.  They should both work and not error out, but only one (the second) should have HUGO ID's.

**Comment by ilan-gold** [comment]
@mccalluc i think those are just ENSEMBL IDs missing a translation so we fall back to those ones

**Comment by ngehlenborg** [comment]
As it should be!

On May 25, 2022, at 4:26 PM, Ilan Gold ***@***.***> wrote:

﻿

@mccalluc<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mccalluc&d=DwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=-zahbihG7P2H0fiPo3RTqce9mFyFku-NJy5aRYNyUSA&m=6yuHYxRBJV5BXWFN3xVc8wYchBLjgfaoUCHCI4qJxlgHKTgREV5SR-uwqkghEp5L&s=G5Imvk0Ir0DSzYGJg3vPAzY2ft398jt-V50Xw9SXwFY&e=> i think those are just ENSEMBL IDs missing a translation so we fall back to those ones

**Pull request merged**